packages/cli/src/lib/log.ts

```diff
@@ -1,25 +1,80 @@
 /* eslint-disable no-console */
 
-let isFirstWorkerReload = true;
+type ConsoleMethod = 'log' | 'warn' | 'error' | 'debug' | 'info';
+const originalConsole = {...console};
+const methodsReplaced = new Set<ConsoleMethod>();
 
+type Matcher = (args: Array<any>) => boolean;
+type Replacer = (args: Array<any>) => void | string[];
+const messageReplacers: Array<[Matcher, Replacer]> = [];
+
+function injectLogReplacer(method: ConsoleMethod) {
+  if (!methodsReplaced.has(method)) {
+    methodsReplaced.add(method);
+    console[method] = (...args: unknown[]) => {
+      const replacer = messageReplacers.find(([matcher]) => matcher(args))?.[1];
+      if (!replacer) return originalConsole[method](...args);
+
+      const result = replacer(args);
+      if (result) return originalConsole[method](...result);
+    };
+  }
+}
+
+injectLogReplacer('log');
+injectLogReplacer('info');
+
+let devMuted = false;
 export function muteDevLogs({workerReload}: {workerReload?: boolean} = {}) {
-  const log = console.log.bind(console);
-  console.log = (first, ...rest) => {
-    // Miniflare logs
-    if (typeof first === 'string' && first.includes('[mf:')) {
+  if (devMuted) return;
+  else devMuted = true;
+
+  let isFirstWorkerReload = true;
+  messageReplacers.push([
+    ([first]) => typeof first === 'string' && first.includes('[mf:'),
+    (args: string[]) => {
+      const first = args[0] as string;
+
       if (workerReload !== false && first.includes('Worker reloaded')) {
-        if (isFirstWorkerReload) isFirstWorkerReload = false;
-        else return log(first.replace('[mf:inf] ', '🔄 ') + '\n', ...rest);
+        if (isFirstWorkerReload) {
+          isFirstWorkerReload = false;
+          // return args as string[];
+          return;
+        }
+
+        return [first.replace('[mf:inf] ', '🔄 ') + '\n', ...args.slice(1)];
       }
 
       if (!first.includes('[mf:err]')) {
         // Hide logs except errors
         return;
       }
-    }
+    },
+  ]);
+}
+
+let authMuted = false;
+export function muteAuthLogs() {
+  if (authMuted) return;
+  else authMuted = true;
 
-    return log(first, ...rest);
-  };
+  messageReplacers.push(
+    [
+      ([first]) => typeof first === 'string' && first.includes('Auto-open'),
+      ([first]) => {
+        return [first.replace(' to Shopify Partners', '')];
+      },
+    ],
+    [
+      ([first]) =>
+        typeof first === 'string' &&
+        (first.includes('Shopify Partners') || first.includes('Logged in')),
```

Comment:
Yeah, it's just that it's hardcoded in a deep function so I think it would be hard for them to make it flexible  -- I assume "Shopify Partners" is correct for most of the other CLIs 🤔